lib/puppet/provider/iis_site/webadministration.rb

```diff
@@ -20,6 +20,8 @@
   def create
     cmd = []
 
+    @resource[:bindings] = sort_bindings(@resource[:bindings])
```

Comment:
Instead of modifying the underlying resource, should you be using the munge method on the type?

Reply:
Or perhaps this should be done in the PowerSHell template instead of in ruby?

https://github.com/puppetlabs/puppetlabs-iis/blob/14ed33d7ba/lib/puppet/provider/templates/webadministration/bindingproperty.ps1.erb

and

https://github.com/puppetlabs/puppetlabs-iis/blob/14ed33d7ba/lib/puppet/provider/templates/webadministration/_getwebsites.ps1.erb#L32-L40

Reply:
e.g.  https://github.com/puppetlabs/puppetlabs-iis/blob/14ed33d7ba/lib/puppet/provider/templates/webadministration/bindingproperty.ps1.erb#L4-L6 would become

``` ruby
Clear-WebConfiguration -Filter "$($website.ItemXPath)/bindings" -Force -ErrorAction Stop
ordered_bindings = resource[:bindings].sort_by { |a| ((a['protocol'] == 'https') ? '0' : '1') + a['bindinginformation'] }
  <%- ordered_bindings.each do |bind| -%>
```

Reply:
> Instead of modifying the underlying resource, should you be using the munge method on the type?

That was my first thought, but from the examples given, the munge method when applied to an array attribute acts on each element of the array, not upon the array as a whole.

Reply:
> e.g. https://github.com/puppetlabs/puppetlabs-iis/blob/14ed33d7ba/lib/puppet/provider/templates/webadministration/bindingproperty.ps1.erb#L4-L6 would become
> 
> ```ruby
> Clear-WebConfiguration -Filter "$($website.ItemXPath)/bindings" -Force -ErrorAction Stop
> 
> <%-
> ordered_bindings = resource[:bindings].sort_by { |a| ((a['protocol'] == 'https') ? '0' : '1') + a['bindinginformation'] }
> 
>   ordered_bindings.each do |bind| -%>
> ```

I'll try that next.

Reply:
Trying it now.